### PR TITLE
Config for RTopGateway

### PR DESCRIPTION
**Pull request by Lys4nder**

### CODE REVIEW

File: RTopGateway/include/config_importer.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#ifndef BACKEND_SERVICE_RTOPGATEWAY_CONFIG_IMPORTER_H
+#define BACKEND_SERVICE_RTOPGATEWAY_CONFIG_IMPORTER_H
+
+#include <string>
+
+namespace ConfigImporter {
```
Comment (oprisor-adrian) [inline RTopGateway/include/config_importer.h:6]: I think the namespace should be the same as the RTopGateway service. The _ConfigImporter_ is part of his implementation.

File: RTopGateway/include/config_importer.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    class ConfigImporter {
+        public:
+            ConfigImporter() = default;
+            void ReadFile();
+            std::string inline GetIp() { return ip_;}
+            int16_t inline GetPort() { return port_;}
+        private:
+            std::string ip_;
+            int16_t port_;
```
Comment (oprisor-adrian) [inline RTopGateway/include/config_importer.h:16]: The port number can take values up to 65535. Consider using a more appropriate data type.
Port (computer networking): https://en.wikipedia.org/wiki/Port_(computer_networking)

File: RTopGateway/include/config_importer.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#ifndef BACKEND_SERVICE_RTOPGATEWAY_CONFIG_IMPORTER_H
+#define BACKEND_SERVICE_RTOPGATEWAY_CONFIG_IMPORTER_H
+
+#include <string>
+
+namespace RTopGateway {
+
+    class ConfigImporter {
```
Comment (oprisor-adrian) [inline RTopGateway/include/config_importer.h:8]: Try to add comments at your class. Also, you can write an example of how I can use the class.
Style guide: https://google.github.io/styleguide/cppguide.html#Class_Comments

File: RTopGateway/include/config_importer.h [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        public:
+            ConfigImporter() = default;
+            void ReadFile();
+            std::string inline GetIp() { return ip_;}
+            int16_t inline GetPort() { return port_;}
+        private:
+            std::string ip_;
+            uint16_t port_;
```
Comment (oprisor-adrian) [inline RTopGateway/include/config_importer.h:16]: Take a look at function comments and variable comments:
- https://google.github.io/styleguide/cppguide.html#Function_Comments
- https://google.github.io/styleguide/cppguide.html#Variable_Comments

Comment (Lys4nder) [reply RTopGateway/include/config_importer.h:16]: As per Google's Style guide "Declaration comments describe use of the function **(when it is non-obvious)**" and "In general the actual name of the **variable should be descriptive enough to give a good idea of what the variable is used for.**", therefore, I genuinely don't think there's a need of comments for such a small class.

File: RTopGateway/include/config_importer.h (hunk 3 above)
Comment (Lys4nder) [reply RTopGateway/include/config_importer.h:8]: As per Google's Style Guide: "Every **non-obvious** class or struct declaration **should** have an accompanying comment that describes what it is for and how it should be used."

**Review by CristianIonutIonescu — COMMENTED** [review]
Although the implementation is good and straightforward, I would suggest to look over Singleton pattern and use it for ConfigImporter. It might be overkill at this point but it's a good exercise to do it.
PS: pay attention to singleton pattern in multithreaded environment

File: RTopGateway/include/config_importer.h [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    class ConfigImporter {
+        public:
+            // this is the static method that controls the access to the singleton instance
+            // on instatiation, it creates the singleton object in a static field, then on
+            // re-runs, it returns the statically stored existing object
+            static ConfigImporter& GetInstance()
+            {
+                static ConfigImporter instance;
+                return instance;
+            }
```
Comment (CristianIonutIonescu) [inline RTopGateway/include/config_importer.h:18]: this isn't how a singleton pattern works. read more about it and come with a proper solution

**Review by CristianIonutIonescu — CHANGES_REQUESTED** [review]
Singleton pattern not correctly implemented

**Pull request merged**